tests/test_utilites.py

```diff
@@ -0,0 +1,421 @@
+"""
+==============
+test_utilities.py
+==============
+
+Test the net2cog utilites functionality.
+"""
+
+import pytest
+from net2cog.netcdf_convert import Net2CogError
+from net2cog.utilities import (
+    resolve_relative_path,
+    is_variable_in_datatree,
+    reorder_dimensions,
+)
+
+
+def test_resolve_relative_path(input_datatree):
+    """Ensure a relative path can be qualified to a full path using the
+    location of the dataset making the reference.
+
+    Tree structure in input_datatree:
+
+    |- science_one(lat, lon)
+    |- lat(lat)
+    |- lon(lon)
+    |- group_one
+       |- science_two(lat, lon)
+       |- science_three(lat, lon)
+       |- group_two
+          | science_four(lat, lon)
+    |- group_five
+       |- variable_one(attr: grid_mapping)
+
+    """
+
+    test_args = [
```

Comment:
Some of this could probably be converted to a [parametrized test](https://docs.pytest.org/en/latest/example/parametrize.html#) but this achieves similar results